app/controllers/course/assessment/question/text_responses_controller.rb

```diff
@@ -24,9 +28,18 @@ def create
 
   def edit
     @question_assessment = load_question_assessment_for(@text_response_question)
+    build_at_least_one_group_one_point if @text_response_question.comprehension_question?
   end
 
   def update
+    if @text_response_question.comprehension_question?
+      @text_response_question.groups.map do |group|
+        group.points.map do |point|
+          point.solutions.map(&:solution_will_change!)
+        end
+      end
+    end
+
     if @text_response_question.update_attributes(text_response_question_params)
```

Comment:
When this fails, `render 'edit'` will fail because `@question_assessment` in not loaded.

Reply:
I have added `@question_assessment = load_question_assessment_for(@text_response_question)` before the `render 'edit'` line.